provisioner/ibm-s3fs-provisioner.go

```diff
@@ -246,11 +247,18 @@ func (p *IBMS3fsProvisioner) validateAnnotations(ctx context.Context, options co
 		}
 	}
 
-	if pvc.SecretNamespace == "" {
-		if sc.SecretNamespace != "" {
-			pvc.SecretNamespace = sc.SecretNamespace
-		} else {
-			pvc.SecretNamespace = options.PVC.Namespace
+	contextLogger.Info(pvcName + ":" + clusterID + " AllowSecretNs: " + strconv.FormatBool(*AllowSecretNs))
+
+	if AllowSecretNs != nil && *AllowSecretNs == false {
+		contextLogger.Info(pvcName + ":" + clusterID + " AllowSecretNs is set to false, the secret will be looked for in same namespace where pvc is created")
+		pvc.SecretNamespace = options.PVC.Namespace
```

Comment:
if (pvc.SecretNamespace != "") -> add warning saying "ibm.io/secret-namespace annotation even if set will be ignored"